### PR TITLE
cd: add GoCD deployment pipelines

### PR DESCRIPTION
**Pull request by joshuarli**
This adds deployment configurations for GoCD, the new internal continuous deployment tool. Please refer to the [Freight Migration Guide](https://www.notion.so/sentry/Freight-Migration-Guide-2bd47bb683254077ad31e7ad625345ae) for info on how to coordinate GoCD deploys during this transitional phase.

### CODE REVIEW

File: .github/workflows/lint-pipelines.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# gocd-cli does not catch all errors, but does catch some simple issues.
+# A better solution may be: https://github.com/GaneshSPatil/gocd-mergeable
+
+echo "GoCD YAML Linting"
+
+find "gocd" -name "*.yaml" -type f -print0 | \
+  xargs -0 -I'{}' bash -c 'printf  "\n🔎 Linting {}\n\t" && gocd-cli configrepo syntax --yaml --raw "{}"'
```
Comment (evanh) [inline .github/workflows/lint-pipelines.sh:9]: Is this meant to lint just the gocd files? What exactly is this doing?

File: gocd/pipelines/snuba-stable.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                branch: master
+                destination: snuba
+        stages:
+            - checks:
+                  approval:
+                      type: manual
+                  fetch_materials: true
+                  jobs:
+                      checks:
+                          timeout: 1800
```
Comment (evanh) [inline gocd/pipelines/snuba-stable.yaml:30]: Is this in seconds? For the entire CI run?

File: gocd/pipelines/snuba-stable.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                          elastic_profile_id: snuba
+                          tasks:
+                              - script: |
+                                    /devinfra/scripts/checks/githubactions/checkruns.py \
```
Comment (evanh) [inline gocd/pipelines/snuba-stable.yaml:34]: Where does this script live? Does it change how these checks are shown in Github?

Comment (evanh) [reply gocd/pipelines/snuba-stable.yaml:34]: Also, why are these checks part of the deployment pipeline? They get run automatically on the master branch whenever we merge. Does this mean they all need to be run again before we can deploy?

**Review by dbanda — APPROVED** [review]

File: .github/workflows/lint-pipelines.sh (hunk 1 above)
Comment (joshuarli) [reply .github/workflows/lint-pipelines.sh:9]: Yeah, it is.

File: gocd/pipelines/snuba-stable.yaml (hunk 2 above)
Comment (joshuarli) [reply gocd/pipelines/snuba-stable.yaml:30]: No, just for the `checks` job.

File: gocd/pipelines/snuba-stable.yaml (hunk 3 above)
Comment (joshuarli) [reply gocd/pipelines/snuba-stable.yaml:34]: https://github.com/getsentry/devinfra-deployment-service/tree/main/gocd_agent/scripts/checks/githubactions

This script checks for the statuses, it doesn't run them.

**Pull request merged**